radixdlt-core/radixdlt/src/main/java/com/radixdlt/LedgerRecoveryModule.java

```diff
@@ -85,33 +134,34 @@ private VerifiedVertexStoreState vertexStoreState(
 		Hasher hasher
 	) {
 		return ledgerEntryStore.loadLastVertexStoreState()
-				.map(serializedVertexStoreState -> {
-					UnverifiedVertex root = serializedVertexStoreState.getRoot();
-					HashCode rootVertexId = hasher.hash(root);
-					VerifiedVertex verifiedRoot = new VerifiedVertex(root, rootVertexId);
+			.filter(vertexStoreState -> vertexStoreState.getHighQC().highestQC().getEpoch() == lastEpochProof.getEpoch() + 1)
+			.map(serializedVertexStoreState -> {
+				UnverifiedVertex root = serializedVertexStoreState.getRoot();
+				HashCode rootVertexId = hasher.hash(root);
+				VerifiedVertex verifiedRoot = new VerifiedVertex(root, rootVertexId);
 
-					ImmutableList<VerifiedVertex> vertices = serializedVertexStoreState.getVertices().stream()
-							.map(v -> new VerifiedVertex(v, hasher.hash(v)))
-							.collect(ImmutableList.toImmutableList());
+				ImmutableList<VerifiedVertex> vertices = serializedVertexStoreState.getVertices().stream()
+					.map(v -> new VerifiedVertex(v, hasher.hash(v)))
+					.collect(ImmutableList.toImmutableList());
 
-					return VerifiedVertexStoreState.create(
-							serializedVertexStoreState.getHighQC(),
-							verifiedRoot,
-							vertices,
-							serializedVertexStoreState.getHighestTC()
-					);
-				})
-				.orElseGet(() -> {
-					UnverifiedVertex genesisVertex = UnverifiedVertex.createGenesis(lastEpochProof.getRaw());
-					VerifiedVertex verifiedGenesisVertex = new VerifiedVertex(genesisVertex, hasher.hash(genesisVertex));
-					LedgerHeader nextLedgerHeader = LedgerHeader.create(
-							lastEpochProof.getEpoch() + 1,
-							View.genesis(),
-							lastEpochProof.getAccumulatorState(),
-							lastEpochProof.timestamp()
-					);
-					QuorumCertificate genesisQC = QuorumCertificate.ofGenesis(verifiedGenesisVertex, nextLedgerHeader);
-					return VerifiedVertexStoreState.create(HighQC.from(genesisQC), verifiedGenesisVertex, Optional.empty());
-				});
+				return VerifiedVertexStoreState.create(
+					serializedVertexStoreState.getHighQC(),
+					verifiedRoot,
+					vertices,
+					serializedVertexStoreState.getHighestTC()
+				);
+			})
+			.orElseGet(() -> {
+				UnverifiedVertex genesisVertex = UnverifiedVertex.createGenesis(lastEpochProof.getRaw());
```

Comment:
I think it's better to move this code into dedicated method.
Also, wider adoption of 'var' will reduce the number of repetitions of type names.